### PR TITLE
Fixed windows build

### PR DESCRIPTION
**Pull request by maths22**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -70,17 +72,28 @@ file(GLOB TINYGETTEXT_HEADERS RELATIVE ${CMAKE_CURRENT_SOURCE_DIR} include/tinyg
...
 ## SDL/iconv usage
 option(HAVE_SDL "Use SDL_iconv instead of system iconv" FALSE)
 if(HAVE_SDL)
 set_source_files_properties(${TINYGETTEXT_SOURCES} PROPERTIES COMPILE_DEFINITIONS HAVE_SDL)
-TARGET_INCLUDE_DIRECTORIES(tinygettext PUBLIC ${SDL2_INCLUDE_DIRS})
+INCLUDE_DIRECTORIES(tinygettext PUBLIC ${SDL2_INCLUDE_DIRS})
 TARGET_LINK_LIBRARIES(tinygettext PUBLIC ${SDL2_LIBRARIES})
 
-else(HAVE_SDL)
+endif(HAVE_SDL)
```
Comment (Mathnerd314) [inline CMakeLists.txt]: The whole point of this is to link to SDL _or_ iconv; tinygettext does not require both. So why is this here?

And what is wrong with TARGET_INCLUDE_DIRECTORIES? You left it alone in [line 87](https://github.com/SuperTux/tinygettext/pull/1/files#diff-af3b638bc2a3e6c650974192a53c7291R87).

File: CMakeLists.txt [hunk 2]
```diff
@@ -105,52 +118,77 @@ else(HAVE_ICONV_CONST)
...
 
-add_definitions(-std=c++0x -O3 -Wall -Wextra -Weffc++ -pedantic)
+if ("${CMAKE_VERSION}" VERSION_GREATER 3.0.99)
+  set_property(GLOBAL PROPERTY CXX_STANDARD 11)
+else ("${CMAKE_VERSION}" VERSION_GREATER 3.0.99)
+  # http://www.guyrutenberg.com/2014/01/05/enabling-c11-c0x-in-cmake/
+  include(CheckCXXCompilerFlag)
+  CHECK_CXX_COMPILER_FLAG("-std=c++11" COMPILER_SUPPORTS_CXX11)
+  if(COMPILER_SUPPORTS_CXX11)
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
```
Comment (Mathnerd314) [inline CMakeLists.txt:140]: You have included a giant compiler flag library, the least you could do is use ADD_NEW_FLAG here for consistency.

File: src/iconv.cpp [hunk 3]
```diff
@@ -110,7 +110,7 @@ IConv::convert(const std::string& text)
 
     // We try to avoid to much copying around, so we write directly into
     // a std::string
-    tinygettext_ICONV_CONST char* inbuf = const_cast<char*>(&text[0]);
+    const char* inbuf = const_cast<char*>(&text[0]);
```
Comment (Mathnerd314) [inline src/iconv.cpp:113]: There is a whole #ifdef section for this in https://github.com/maths22/tinygettext/blob/master/include/tinygettext/iconv.hpp#L25, you should fix it there. Which iconv are you actually using?

File: CMakeLists.txt (hunk 2 above)
Comment (Mathnerd314) [inline CMakeLists.txt:140]: I think just setting  cmake_minimum_required to 3.1 would work too; Travis is using CMake 3.3, and any decent development system will have a recent CMake.

File: CMakeLists.txt (hunk 1 above)
Comment (maths22) [inline CMakeLists.txt]: I will re-work this pull request and resubmit it.  This was actually done (apparently) because when it attempts to build the tests, HAVE_SDL is unset, forcing it to try to use iconv.  I believe that was my reasoning for the TARGET_INCLUDE_DIRECTORIES thing as well.

File: CMakeLists.txt (hunk 2 above)
Comment (maths22) [inline CMakeLists.txt:140]: I will admit I should have done that-I didn't because I pulled that block of code straight from the SuperTux CMakeFiles (which I should actually convert to use ADD_NEW_FLAG).  The SuperTux code actually only used the bottom version; I added the top one because it is more robust across compilers.  Increasing CMAKE_MINIMUM_REQUIRED to 3.1 would seem reasonable; however, it will also force people to compile their own CMake on some platforms, like Ubuntu (the repo for 15.04 has 3.0) [I would still do it, but it is worth considering]

File: src/iconv.cpp (hunk 3 above)
Comment (maths22) [inline src/iconv.cpp:113]: I will look that over.  I am using the win32 gnu version of ICONV, but I think the combination of that and SDL messed things up.  That said, what is the issue with it always being a const?

Comment (Mathnerd314) [inline src/iconv.cpp:113]: See https://sourceware.org/bugzilla/show_bug.cgi?id=2962 (featuring Ulrich Drepper) and [1](http://stackoverflow.com/questions/11421439/how-can-i-portably-call-a-c-function-that-takes-a-char-on-some-platforms-and) [2](http://pubs.opengroup.org/onlinepubs/9699919799/functions/iconv.html) [3](https://www.gnu.org/savannah-checkouts/gnu/libiconv/documentation/libiconv-1.14/iconv.3.html). The short answer is that some systems use const, some don't, they are unlikely to standardize, and using the wrong version results in a compilation error.
